### PR TITLE
Added labels for classical-ml

### PR DESCRIPTION
**Pull request by sharvil10**
## Description

This PR adds labels to Classical ML services for support matrix docs.

## Changes Made

1.  Adds labels in all services in the file `classical-ml/docker-compose.yaml`

- [x] The code follows the project's [coding standards](../CONTRIBUTING.md#code-style).
- [x] No Intel Internal IP is present within the changes.
- [x] The documentation has been updated to reflect any changes in functionality.

## Validation

- [x] I have tested any changes in container groups locally with [`test_runner.py`](../test-runner/README.md) with all existing tests passing, and I have added new tests where applicable.

### CODE REVIEW

File: classical-ml/docker-compose.yaml [hunk 1]
```diff
@@ -33,6 +33,14 @@ services:
         REGISTRY: ${REGISTRY}
         REPO: ${REPO}
       context: .
+      labels:
+        dependency.python: ${PYTHON_VERSION:-3.10}
+        dependency.python.pip: requirements.txt
+        docs: classical-ml
+        org.opencontainers.base.name: "intel/python:3.10-core"
+        org.opencontainers.image.name: "intel/intel-optimized-ml"
+        org.opencontainers.image.title: "Intel® Extension for Scikit-learn, XGBoost Base Image"
```
Comment (tylertitsworth) [inline classical-ml/docker-compose.yaml:42]:
```suggestion
        org.opencontainers.image.title: "Intel® Optimized ML Base Image"
```

File: classical-ml/docker-compose.yaml [hunk 2]
```diff
@@ -42,6 +50,11 @@ services:
     pull_policy: always
   jupyter:
     build:
+      labels:
+        dependency.python.pip: jupyter-requirements.txt
+        org.opencontainers.base.name: "intel/intel-optimized-ml:${SKLEARN_VERSION:-2024.4.0}-${PACKAGE_OPTION:-pip}-base"
+        org.opencontainers.image.title: "Intel® Extension for Scikit-learn, XGBoost Base Image"
```
Comment (tylertitsworth) [inline classical-ml/docker-compose.yaml:56]:
```suggestion
        org.opencontainers.image.title: "Intel® Optimized ML Jupyter Image"
```

**Review by tylertitsworth — CHANGES_REQUESTED** [review]

File: classical-ml/docker-compose.yaml (hunk 2 above)
Comment (sharvil10) [reply classical-ml/docker-compose.yaml]: Updated

File: classical-ml/docker-compose.yaml (hunk 1 above)
Comment (sharvil10) [reply classical-ml/docker-compose.yaml]: Updated

**Review by tylertitsworth — DISMISSED** [review]

**Pull request merged**